Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs

```diff
@@ -0,0 +1,583 @@
+
+        /// <summary>
+        /// This is the radius of either the Cylinder or Sphere mapping and is ignored when using the plane mapping.
+        /// </summary>
+        [Range(0.05f, 5.0f)]
+        [Tooltip("Radius for the sphere or cylinder")]
+        public float Radius = 2f;
+
+        /// <summary>
+        /// Number of rows per column, column number is automatically determined
+        /// </summary>
+        [Tooltip("Number of rows per column")]
+	    public int Rows = 3;
+
+        /// <summary>
+        /// Width of the cell per object in the collection.
+        /// </summary>
+        [Tooltip("Width of cell per object")]
+        public float CellWidth = 0.5f;
+
+        /// <summary>
+        /// Height of the cell per object in the collection.
+        /// </summary>
+        [Tooltip("Height of cell per object")]
+        public float CellHeight = 0.5f;
+
+        /// <summary>
+        /// Reference mesh to use for rendering the sphere layout
+        /// </summary>
+        [HideInInspector]
+        public Mesh SphereMesh;
+
+        /// <summary>
+        /// Reference mesh to use for rendering the cylinder layout
+        /// </summary>
+        [HideInInspector]
+        public Mesh CylinderMesh;
+        #endregion
+
+
+        #region private variables
+        private int _columns;
+        private float _width;
+        private float _height;
+        private float _circ;
+        private Vector2 _halfCell;
+        #endregion
+
+        public float Width
+        {
+            get { return _width; }
+        }
+
+        public float Height
+        {
+            get { return _height; }
+        }
+
+        /// <summary>
+        /// Update collection is called from the editor button on the inspector.
+        /// This function rebuilds / updates the layout.
+        /// </summary>
+        public void UpdateCollection()
+        {
+            // Check for empty nodes and remove them
+            List<CollectionNode> emptyNodes = new List<CollectionNode>();
+
+            for (int i = 0; i < NodeList.Count; i++)
+            {
+                if (NodeList[i].transform == null || (IgnoreInactiveTransforms && !NodeList[i].transform.gameObject.activeSelf))
+                {
+                    emptyNodes.Add(NodeList[i]);
+                }
+            }
+
+            // Now delete the empty nodes
+            foreach (CollectionNode node in emptyNodes)
+            {
+                NodeList.Remove(node);
+            }
+
+            emptyNodes.Clear();
+
+            // Check when children change and adjust
+            for (int i = 0; i < this.transform.childCount; i++)
+            {
+                Transform child = this.transform.GetChild(i);
+
+                if (!ContainsNode(child) && (child.gameObject.activeSelf || !IgnoreInactiveTransforms))
+                {
+                    CollectionNode node = new CollectionNode();
+
+                    // TODO MICROSOFT: Initial offset
```

Comment:
What's this TODO for?

Reply:
I'm removing the TODO's that are in there for now.  Some are more opinion than actionable todos anyway.